profiling/profiler.go

```diff
@@ -0,0 +1,46 @@
+package profiling
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	hpprof "net/http/pprof"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// Profiler runs and controls a HTTP pprof interface.
+type Profiler struct {
+	server *http.Server
+}
+
+// NewProfiler creates an instance of the profiler with
+// the given port.
+func NewProfiler(port int) *Profiler {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/debug/pprof/", hpprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", hpprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", hpprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", hpprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", hpprof.Trace)
+	p := Profiler{
+		server: &http.Server{
+			Addr:    fmt.Sprintf(":%d", port),
+			Handler: mux,
+		},
+	}
+	return &p
+}
+
+// Run starts the HTTP pprof in the background.
+func (p *Profiler) Run() {
+	go func() {
+		log.Info("debug server stopped", "err", p.server.ListenAndServe())
+	}()
```

Comment:
following a discussion we had in the past, what do you think about moving the concurrency part to the main function? So that this one as a library is synchronous and it's the main function that runs a different go routine?

Reply:
I think it's a bit a different case. Profiler is expected to run in the background, it's a diagnostic tool, not the business logic.

Reply:
That has been the intention. But the comment showed my approach with git to cleanup the sol problem dropped also the server starting. 🤕